akka-actor/src/main/scala/akka/actor/ActorSystem.scala

```diff
@@ -82,11 +82,28 @@ object BootstrapSetup {
 
 }
 
-abstract class ProviderSelection private (private[akka] val identifier: String)
+/**
+ * @param identifier the simple name of the selected provider
+ * @param fqcn the fully-qualified class name of the selected provider
+ */
+abstract class ProviderSelection private (private[akka] val identifier: String, private[akka] val fqcn: String) {
+
+  /** INTERNAL API */
+  @InternalApi private[akka] def hasCluster: Boolean =
+    this match {
+      case ProviderSelection.Cluster => true
+      case _                         => false
```

Comment:
hmm, for `Local`, `Cluster` and `Remote` we know the answer, but it seems a bit overconfident to assume all 'customer' actor selections are non-clustered... perhaps we should make that a field on `Custom`?

Reply:
`private[akka] class ClusterActorRefProvider` so we don't support any custom subclasses of it

In practise I would say that we don't support any custom ActorRefProvider, and it has `@DoNotInherit trait ActorRefProvider`

In other words, this logic is ok